src/Uno.UI/Extensions/PermissionsHelper.Android.cs

```diff
@@ -19,11 +19,185 @@ public static void Initialize()
 					.CreateAsync<string, bool>(TryGetPermissionCore)
 					.LockInvocation(InvocationLockingMode.Share);
 
-			Windows.Extensions.PermissionsHelper.Initialize(getPermission, CheckPermission);
-		}
+			Windows.Extensions.PermissionsHelper.Initialize(getPermission, CheckPermission, MissingPermissions, TryAskPermissionAsync);
+		}
+
+		private static int _permissionRequest;
+		#region "submethods for MissingPermissions"
+		private static IList<string> GetManifestPermissions()
+		{
+			// get all permission declared in Manifest (can be null)
+			Android.Content.Context context = Android.App.Application.Context;
+			Android.Content.PM.PackageInfo packageInfo =
+				context.PackageManager.GetPackageInfo(context.PackageName, Android.Content.PM.PackageInfoFlags.Permissions);
+			return packageInfo?.RequestedPermissions;
+		}
+
+		private static bool AreAllPermissionDeclared(IList<string> manifestPermissions, string[] requiredPermissions)
+		{
+			// check if all requiredPermissions are declared in Manifest 
+			if (requiredPermissions is null)
+			{
+				return true;    // no required permissions, so - everything is OK
+			}
+
+			foreach (string permission in requiredPermissions)
+			{
+				if (!manifestPermissions.Any(p => p.Equals(permission, StringComparison.OrdinalIgnoreCase)))
+				{
+					return false;
+				}
+			}
+
+			return true;
+
+		}
+
+		private static List<string> PermissionWeWant(IList<string> manifestPermissions, string[] requiredPermissions, string[] optionalPermissions)
+		{
+			// prepare list of all permissions
+			var allPermissions = new List<string>();
+			if (requiredPermissions != null)
+				allPermissions.AddRange(requiredPermissions.ToList());
+
+			// add all optional permission, found in Manifest
+			if (optionalPermissions != null)
+			{
+				foreach (string permission in optionalPermissions)
+				{
+					if (manifestPermissions.Any(p => p.Equals(permission, StringComparison.OrdinalIgnoreCase)))
+					{
+						allPermissions.Add(permission);
+					}
+				}
+			}
+
+			return allPermissions;
+		}
+
+		private static List<string> PermissionToAsk(IList<string> allPermissions)

```

Comment:
```suggestion
		private static List<string> GetPermissionsToRequest(IList<string> allPermissions)

```

Reply:
General comment about "Ask" and "Request" (in all methods) - we doesn't request permission, we kindly ask user for granting us permission...

Reply:
`Request` is what the android API is called, if I'm not mistaken, this is only aligning.